lib/perl/Genome/InstrumentData/AlignmentResult/Merged.pm

```diff
@@ -297,6 +368,15 @@ sub collect_individual_alignments {
             };
         }
 
+        my $test_name;
+        if ($ENV{GENOME_SOFTWARE_RESULT_TEST_NAME}) {
+            $test_name = $ENV{GENOME_SOFTWARE_RESULT_TEST_NAME};
+        } elsif ($self->test_name) {
```

Comment:
In fact, in get_merged_alignment_results on AlignmentResult, you always use the test_name on the current result.